### PR TITLE
Workaround babel startsWith not being transpiled in some case

### PR DESCRIPTION
**Pull request by noirbizarre**
IE does not have natively `String.prototype.startsWith`.
Given how Babel is working when using babel-runtime, polyfills are not applied to prototype and there is some case where babel-runtime is not able to detected variable type and to transpile the polyfill.

This PR fix this on `variable.startsWith`.

### CODE REVIEW

File: js/dataset/display.js [hunk 1]
```diff
@@ -54,10 +59,10 @@ function prepare_resources() {
                 const url = $self.parent().property('url').first().attr('href');
                 const group = $Dataset.property('alternateName').value(); // This is the slug.
 
-                if (!url.startsWith(window.location.origin)
+                if (!startsWith(url, window.location.origin)
                 // TODO: temporary fix before we move all statics on the same server
                     && !url.match(/:\/\/(.[^/]+)/)[1].endsWith('data.gouv.fr')) {
```
Comment (vinyll) [inline js/dataset/display.js]: Does `startWith` not work and `endWith` works properly?

Comment (noirbizarre) [inline js/dataset/display.js]: Yes, in this case: there is the `match` before (native on IE) which is probably hinting babel on the resulting type.
But you're right, I might be better to be homogeneous on this one.

**Comment by vinyll** [comment]
r+wc

**Comment by noirbizarre** [comment]
Babel-runtime is not a strict equivalent of babel-polyfill:
- https://phabricator.babeljs.io/T892: `Array.prototype.find` doesn't work in the runtime
- https://phabricator.babeljs.io/T6905: babel-plugin-transform-runtime: handle methods differently?

In our case, each time a function call an expected prototype method only existing in ES6, it fails in IE which does not yet implement most of then.

Conclusion: I will update this PR to use `babel-polyfill` when possible

**Comment by noirbizarre** [comment]
And this article explain the difference between `babel-polyfill` and `babel-runtime`/`babel-runtime-transform-plugin`: https://medium.com/@jcse/clearing-up-the-babel-6-ecosystem-c7678a314bf3

**Comment by davidbgk** [comment]
I'm surprised we have to import `babel-polyfill` in all these files, as I understand it that should only be required in root JS files?

**Comment by noirbizarre** [comment]
Yes, it's imported only in root JS files, but we have one root file for admin + one for each view.

**Comment by davidbgk** [comment]
Ah, makes sense now, I forgot that we have so many root files, my bad. I'm concerned about the fact that we may forgot to add that line for new root files, how can we prevent this?

**Comment by noirbizarre** [comment]
@davidbgk we need to try on IE11 too. I'm using https://dev.windows.com/en-us/microsoft-edge/tools/vms/linux/ to do so.

If it works without, it meeds we don't use ES6 specifics so the forgot is not bad in this case.

Or, we can provide a template root file.

Perosnnaly, I think, we should as an objective, reduce the number of root files :)

**Comment by davidbgk** [comment]
> Perosnnaly, I think, we should as an objective, reduce the number of root files :)

It depends, with HTTP/2 it's not a bad practice anymore to be modular ;)

**Comment by noirbizarre** [comment]
@davidbgk Next step: HTTP2, Python 3, ipv6 ? :)

**Comment by davidbgk** [comment]
Served by GNU Hurd :p

**Comment by noirbizarre** [comment]
:)

After looking, switching to http2 is simply adding a keyword on NGinx, the real question is between reverse-proxy. I don't know if it's possible to have HTTP2 -> NGINX/HAProxy -> HTTP -> Application.

**Comment by noirbizarre** [comment]
so, Can I merge this one ?

**Comment by davidbgk** [comment]
r+, we have to remember that `.babelrc` is not anymore in use on the `dev` branch and update the `webpack.config` accordingly during the merge.

**Comment by noirbizarre** [comment]
@davidbgk yep, it will conflict on merge as a reminder ;)

**Pull request merged**